nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/test/java/org/apache/nifi/json/TestJsonTreeRowRecordReader.java

```diff
@@ -1062,38 +1026,197 @@ public void testChoseSuboptimalSchemaWhenDataHasExtraFields() throws Exception {
+        String jsonPath = "src/test/resources/json/single-element-nested.json";
+
+        SimpleRecordSchema expectedRecordSchema = new SimpleRecordSchema(Arrays.asList(
+                new RecordField("id", RecordFieldType.INT.getDataType()),
+                new RecordField("balance", RecordFieldType.DOUBLE.getDataType())
+        ));
+
+        List<Object> expected = Collections.singletonList(
+                new MapRecord(expectedRecordSchema, new HashMap<String, Object>() {{
+                    put("id", 42);
+                    put("balance", 4750.89);
+                }})
+        );
+
+        testReadRecords(jsonPath, expected, "name");
+    }
+
+    @Test
+    void testSkipToSimpleFieldAndNoNestedObjectOrArrayFound() throws IOException, MalformedRecordException {
+        String jsonPath = "src/test/resources/json/single-element-nested-array-middle.json";
+
+        List<Object> expected = Collections.emptyList();
+
+        testReadRecords(jsonPath, expected, "name");
+    }
+
+    @Test
+    void testSkipToNonExistentFieldWithDefinedSchema() throws IOException, MalformedRecordException {
+        String jsonPath = "src/test/resources/json/single-element-nested-array-middle.json";
+
+        SimpleRecordSchema expectedRecordSchema = new SimpleRecordSchema(getDefaultFields());
+        List<Object> expected = Collections.emptyList();
+
+        Exception exception = assertThrows(IOException.class,
+                () -> testReadRecords(jsonPath, expectedRecordSchema, expected, "notfound")
+        );
+
+        assertEquals("The defined skipTo json field is not found when processing json as NiFi record.", exception.getMessage());
+        assertEquals(AbstractJsonRowRecordReader.class.getName(), exception.getStackTrace()[0].getClassName());
+    }
+
     private void testReadRecords(String jsonPath, List<Object> expected) throws IOException, MalformedRecordException {
         // GIVEN
         final File jsonFile = new File(jsonPath);
 
         try (
-            InputStream jsonStream = new ByteArrayInputStream(FileUtils.readFileToByteArray(jsonFile));
+            InputStream jsonStream = new ByteArrayInputStream(FileUtils.readFileToByteArray(jsonFile))
         ) {
-            RecordSchema schema = inferSchema(jsonStream);
+            RecordSchema schema = inferSchema(jsonStream, null);
 
             // WHEN
             // THEN
             testReadRecords(jsonStream, schema, expected);
         }
     }
 
+    private void testReadRecords(String jsonPath, List<Object> expected, String skipToField) throws IOException, MalformedRecordException {
+        // GIVEN
+        final File jsonFile = new File(jsonPath);
+
+        try (
+                InputStream jsonStream = new ByteArrayInputStream(FileUtils.readFileToByteArray(jsonFile))
+        ) {
+            RecordSchema schema = inferSchema(jsonStream, skipToField);
+
+            // WHEN
+            // THEN
```

Comment:
It is best to avoid introducing the `GIVEN/WHEN/THEN` comments since they do not provide substantive detail on how the code works.